extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -0,0 +1,371 @@
+      } else {
+        user().error(TAG, 'Invalid result for class binding', newValue);
+      }
+    } else {
+      if (newValue === true) {
+        element.setAttribute(property, '');
+      } else if (newValue === false) {
+        element.removeAttribute(property);
+      } else {
+        const oldValue = element.getAttribute(property);
+        if (oldValue === newValue) {
+          return;
+        }
+
+        /** @type {boolean|number|string|null} */
+        const attributeValue = this.attributeValueOf_(newValue);
+        if (attributeValue === null) {
+          user().error(TAG, 'Invalid result for attribute binding', newValue);
+          return;
+        }
+
+        // TODO(choumx): Does `newValue` pass validator value_casei,
+        // value_regex, blacklisted_value_regex, value_url>allowed_protocol,
+        // mandatory?
+
+        element.setAttribute(property, attributeValue);
+
+        // Update internal state for AMP elements.
+        if (element.classList.contains('-amp-element')) {
+          const resources = element.getResources();
+          if (property === 'width') {
+            user().assert(typeof attributeValue === 'number',
+                'Invalid result for [width]: %s', attributeValue);
+            resources./*OK*/changeSize(element, undefined, attributeValue);
+          } else if (property === 'height') {
+            user().assert(typeof attributeValue === 'number',
+                'Invalid result for [height]: %s', attributeValue);
+            resources./*OK*/changeSize(element, attributeValue, undefined);
+          }
+          element.attributeChangedCallback(property, oldValue, attributeValue);
+        }
+      }
+    }
+  }
+
+  /**
+   * If the current value of `binding` equals `expectedValue`, returns true.
+   * Otherwise, returns false.
+   * @param {!BindingDef} binding
+   * @param {BindExpressionResultDef} expectedValue
+   * @private
+   */
+  verifyBinding_(binding, expectedValue) {
+    const property = binding.property;
+    const element = binding.element;
+
+    // TODO(choumx): Support objects for attributes.
+
+    let initialValue;
+    let match = true;
+
+    if (property === 'text') {
+      initialValue = element.textContent;
+      match = (initialValue.trim() === expectedValue.trim());
+    } else if (property === 'class') {
+      initialValue = element.classList;
+
+      /** @type {!Array<string>} */
+      let classes = [];
+      if (Array.isArray(expectedValue)) {
+        classes = expectedValue;
+      } else if (typeof expectedValue === 'string') {
+        classes = expectedValue.split(' ');
+      } else {
+        user().error(TAG,
+            'Unsupported result for class binding', expectedValue);
+      }
+      match = this.compareStringArrays_(initialValue, classes);
+    } else {
+      const attribute = element.getAttribute(property);
+      if (typeof expectedValue === 'boolean') {
+        initialValue = !!attribute;
```

Comment:
[Continuing](https://github.com/ampproject/amphtml/pull/6449#discussion_r90712825):

You're casting the value from `#getAttribute`, which in this case would be:

- "false", we expect boolean. But `!!"false" === true`.
- "'false'". This one works fine, because we expect a string.